test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/HistoryTests.cs

```diff
@@ -194,52 +194,78 @@ public async Task GivenAValueForSinceAndBeforeWithModifications_WhenGettingSyste
         [Fact]
         public async Task GivenAValueForSinceAndBeforeCloseToLastModifiedTime_WhenGettingSystemHistory_TheServerShouldNotMissRecords()
         {
-            var tag = Guid.NewGuid().ToString();
+            // as this test is timing dependent, we will try it three times
+            // to see if it can pass
+            for (int i = 0; i < 3; i++)
+            {
+                bool success = false;
+                var newResources = new List<Resource>();
 
-            var since = await GetStartTimeForHistoryTest(tag);
+                try
+                {
+                    var tag = Guid.NewGuid().ToString();
 
-            var newResources = new List<Resource>();
+                    var since = await GetStartTimeForHistoryTest(tag);
 
-            // First make a few edits
-            _createdResource.Resource.Text = new Narrative { Div = "<div>Changed by E2E test</div>" };
-            await CreateResourceWithTag(_createdResource.Resource, tag);
+                    // First make a few edits
+                    _createdResource.Resource.Text = new Narrative { Div = "<div>Changed by E2E test</div>" };
+                    await CreateResourceWithTag(_createdResource.Resource, tag);
 
-            newResources.Add(await CreateResourceWithTag(Samples.GetDefaultPatient().ToPoco(), tag));
-            newResources.Add(await CreateResourceWithTag(Samples.GetDefaultOrganization().ToPoco(), tag));
-            Thread.Sleep(1000);
-            newResources.Add(await CreateResourceWithTag(Samples.GetJsonSample("BloodGlucose").ToPoco(), tag));
-            newResources.Add(await CreateResourceWithTag(Samples.GetJsonSample("BloodPressure").ToPoco(), tag));
-            newResources.Add(await CreateResourceWithTag(Samples.GetJsonSample("Patient-f001").ToPoco(), tag));
-            newResources.Add(await CreateResourceWithTag(Samples.GetJsonSample("Condition-For-Patient-f001").ToPoco(), tag));
+                    newResources.Add(await CreateResourceWithTag(Samples.GetDefaultPatient().ToPoco(), tag));
+                    newResources.Add(await CreateResourceWithTag(Samples.GetDefaultOrganization().ToPoco(), tag));
+                    await Task.Delay(1000);
```

Comment:
>await Task.Delay(1000); [](http://example.com/codeflow?start=20&length=23)

I don't understand why this delay is needed

Reply:
This delay is to make sure we have a gap between the edits so we can query just the last 4 edits.

Reply:
When just running the test locally, I have seen the exact same time stamp in milliseconds on all 7 resources, which is why the gap was added.  But, 1 second may be a bit too long.  I will reduce it to 100ms.